### PR TITLE
Draft: Verifiable Credential handler CAIP

### PR DESCRIPTION
**Pull request by bumblefudge**
still very much WIP!

### CODE REVIEW

File: caip-XX.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+<!--The motivation is critical for CAIP. It should clearly explain why the state of the art is inadequate to address the problem that the CAIP solves. CAIP submissions without sufficient motivation may be rejected outright.-->
+Erring on the side of protocol neutrality and making the dapp-wallet connection
+a simple interface that bootstraps to more complex ones was the design goal
+here. It may even be too minimal for many use-cases, but hopefully at least a
+building block for more complex protocols down the line.
+
+## Specification
+<!--The technical specification should describe the standard in detail. The specification should be detailed enough to allow competing, interoperable implementations. -->
+
+```
```
[the remark this answers is not in the document]
Comment (bumblefudge) [reply caip-XX.md:39]: not applicable, just high-level notes to self for now :D

**Comment by awoie** [comment]
for reference, i tried to do something similar a while ago: https://github.com/veramolabs/credential-provider-eip

**Comment by bumblefudge** [comment]
Can't officially request review from people who haven't previously contributed or been added to the CASA org, but @taotao-circle , @mirceanis , @awoie , @andresuribe87 have all been asking about this CAIP so here it is, a working draft!

**Comment by bumblefudge** [comment]
Still todo:
- [ ] error handling (ideally aligned with VC-API error codes?)
- [ ] quick to update examples to presentation exchange v2 syntax and nesting

File: CAIPs/caip-169.md [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+<!--A short (~200 word) description of the technical issue being addressed.-->
+After a dapp has requested and a wallet has declared basic verifiable
+credentials capabilities in the CAIP-25 feature-discovery step, dapps can assume
+the wallet will be able to support all the methods defined below. Note
+that many of these calls should be routable to services, other software, or even
+other dapps, without leaking information about the wallet or architecture to the
+calling dapp.
+
+This CAIP describes three methods that combine to act as a Credential Provider
+(CP) to support _Verifiabe Credentials_ (VCs) storage, issuance, selective
```
Comment (taotao-circle) [inline CAIPs/caip-169.md:42]: typo: Verifiable

Comment (ligi) [inline CAIPs/caip-169.md:42]:
```suggestion
(CP) to support _Verifiable Credentials_ (VCs) storage, issuance, selective
```

File: CAIPs/caip-169.md [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+This proposal defines a set of API extensions that would enable web3 wallet
+providers to offer wallet and credential interactions to web origins that
+already support web3 wallet providers.
+
+This functionality is similar to the interfaces supported by the [credential
+handler api](https://w3c-ccg.github.io/credential-handler-api/), which does not
+support the [Presentation Exchange Data
+Model](https://identity.foundation/presentation-exchange/) specification but
+does offer more flexible support for Linked Data verifiable credential formats
+and a rich interface for wallet selection via the Credential Hander API (CHAPI).
```
Comment (ligi) [inline CAIPs/caip-169.md:301]:
```suggestion
and a rich interface for wallet selection via the Credential Handler API (CHAPI).
```

File: CAIPs/caip-169.md [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+`creds_store`
+
+##### Params:
+
+- `vc` - A Verifiable Credential.
+
+##### Returns:
+
+- `error` - OPTIONAL. If `vc` was malformed or does not comply with the
+  Verifialbe Credentials Profile defined in this specification.
```
Comment (ligi) [inline CAIPs/caip-169.md:149]:
```suggestion
  Verifiable Credentials Profile defined in this specification.
```

File: CAIPs/caip-169.md [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+
+Specifications (Dependencies)
+- [VC spec][] - Core data model for verifiable credentials 
+- [DID EIP][] - Prior art for handling basic DID functions as methods in the
+  standard ethereum provider; these are not required for VC functionality per se
+  but can greatly extend the verification model for issuers and holders of VCs
+  alike, enable on-chain VC verification, etc.
+- [Data Integrity spec][] - Formerly referred to as "Linked Data Proofs," this
+  specification for embedding proofs in structured data objects like JSON-LD
+  documents and verifiable credentialy specifies the handling of distinct
```
Comment (ligi) [inline CAIPs/caip-169.md:371]: verifiable credentials ?

**Review by ligi — CHANGES_REQUESTED** [review]

**Review by mirceanis — COMMENTED** [review]
Looks good so far.

I suggested some fixes for typos

File: CAIPs/caip-169.md [hunk 6]
```diff
@@ -0,0 +1,407 @@
+---
+caip: 169
+title: Simple Credential Interface Provider
+author: Oliver Terbu (@awoie), Mircea Nestor (@mirceanis), Juan Caballero (@bumblefudge)
```
[the remark this answers is not in the document]
Comment (bumblefudge) [reply CAIPs/caip-169.md:4]: embarassing, sorry!

**Comment by awoie** [comment]
@bumblefudge can you make sure it is getting as close as possible to https://openid.net/specs/openid-4-verifiable-presentations-1_0.html

**Comment by bumblefudge** [comment]
No I cannot "make sure" but maybe if Kim helps me we can get reasonably close. Are there test vectors and a stable version I should read before pulling in the Big Hamilton-Duffy guns?

File: CAIPs/caip-169.md [hunk 7]
```diff
@@ -0,0 +1,516 @@
...
+        "ES256K",
+        "ES384"
+    ‌ ]
+  ‌  },
+  ‌  "jwt_vp": {
+      "alg_values_supported": [
+        "ES256K",
+        "EdDSA"
+    ‌ ]
+  ‌ }
```
Comment (bumblefudge) [inline CAIPs/caip-169.md:358]:
```suggestion
   ‌ }
```

File: CAIPs/caip-169.md [hunk 8]
```diff
@@ -0,0 +1,516 @@
...
+algorithm](https://www.iana.org/assignments/jose/jose.xhtml) support of embedded
+and external proofs for VCs.  The abbreviations for common VC formats in
+production today are taken from the [DIF Claim Format registry][] of the
+[Presentation Exchange][] meta-protocol specification governed by the
+Decentralized Identity Foundation, while the Verifiable Credentials profile
+defining their proof formats are specified in the [VC spec][] itself.
+
+#### Embedded Proofs
+
+CPs MUST support the following LD-Proof types for embedded proofs (i.e. VC-LDP):
```
Comment (kimdhamilton) [inline CAIPs/caip-169.md:107]: I didn't see CP defined anywhere, but perhaps it's well known?

File: CAIPs/caip-169.md [hunk 9]
```diff
@@ -0,0 +1,516 @@
...
+
+##### Params:
+
+- `vc` - A Verifiable Credential.
+
+##### Returns:
+
+- `error` - OPTIONAL. If `vc` was malformed or does not comply with the
+  Verifiable Credentials Profile defined in this specification.
+
```
Comment (kimdhamilton) [inline CAIPs/caip-169.md:180]: Does this need to add any detail about expected implementation/behavior of VC verification?

**Comment by kimdhamilton** [comment]
> @bumblefudge can you make sure it is getting as close as possible to https://openid.net/specs/openid-4-verifiable-presentations-1_0.html

@awoie are you referring to the Presentation Exchange objects? Or something else in OpenID 4 VP protocol? In [PEv2](https://identity.foundation/presentation-exchange/#what-is-new) we did a large amount of work with David Chadwick, Kristina, and others to enable OpenID 4 VP / PE alignment.

If PE is the concern, it likely makes more sense for _this_ spec to target PEv2 conformance (if there's any difference between PEv2 and the spec you linked), because PEv2 should be the new target state. 

But let me know if there's something else I'm missing

File: CAIPs/caip-169.md (hunk 8 above)
Comment (bumblefudge) [reply CAIPs/caip-169.md:107]: Whoops, missed that somehow.  Oliver's original draft referred to VC issuance as a "Credential Provider" when this was drafted as a potential EIP; it's fairly standard terminology in the RPC-based dapp<>wallet<>node transport model of EVM

File: CAIPs/caip-169.md [hunk 10]
```diff
@@ -0,0 +1,516 @@
...
+that many of these calls should be routable to services, other software, or even
+other dapps, without leaking information about the wallet or architecture to the
+calling dapp.
+
+This CAIP describes a few core RPC methods that combine to act as a wallet-side
+Credential Provider (CP) to support _Verifiable Credentials_ (VCs) storage,
+issuance, proof-of-control, and even (with some extensions) selective disclosure
+or advancing proofing of those credentials. 
+
+VCs are usually self-certifiable attestations from an issuer about the subject
```
[the remark this answers is not in the document]
Comment (bumblefudge) [reply CAIPs/caip-169.md]: awesome thanks for simplifying i was just staring at that paragraph not sure how to turn it into normal english somehow.  editor's block.

**Review by Wyeth8 — APPROVED** [review]

**Pull request merged**